kotest-property/src/commonMain/kotlin/io/kotest/property/Arbitrary.kt

```diff
@@ -0,0 +1,93 @@
+package io.kotest.property
+
+import io.kotest.property.arbitraries.double
+import io.kotest.property.arbitraries.float
+import io.kotest.property.arbitraries.int
+import io.kotest.property.arbitraries.long
+import kotlin.random.Random
+
+/**
+ * An [Arbitrary] is a type of [Gen] which generates two types of values: [edgecases] and [samples].
+ *
+ * Edge cases are values that are a common source of bugs. For example, a function using ints is
+ * more likely to fail for common edge cases like zero, minus 1, positive 1, [Int.MAX_VALUE] and [Int.MIN_VALUE]
+ * rather than random values like 965489. Therefore, edge cases can be included in sequences
+ * generated by an arbitrary.
+ *
+ * Not all arbitraries will utilize edge cases. For example, if you define an integer generator
+ * using a subset of the number space - say from 100 to 250,000 - then no edge cases are provided.
+ *
+ * Samples are chosen randomly from the sample space and are used to give a greater breadth to
+ * the test cases. For example, in the case of a function using integers, these random values
+ * could be from across the entire integer number line, or could be limited to a subset of ints
+ * such as natural numbers or even numbers.
+ */
+interface Arbitrary<T> : Gen<T> {
+
+   /**
+    * Returns the values that are considered common edge case for the type.
+    *
+    * For example, for Strings this may include the empty string, a string with white space,
+    * a string with unicode, and a string with non-printable characters.
+    *
+    * The result can be empty if for type T there are no common edge cases.
+    *
+    * @return the common edge cases for type T.
+    */
+   fun edgecases(): Iterable<T>
```

Comment:
Maybe make this default to `emptyList`?

Reply:
I didn't want to do that because then people might forget it's there to override.